### ISSUE TITLE
Couldn't get dependency '//.okbuck/keystore:app_keystore' of target '//app:bin'

### ISSUE THREAD

**Issue by wzpan** [Piasy/OkBuck#23]
I got the following error after I execute `buck run app`

```
➜  appframework git:(master) ✗ buck run app       
Using buckd.
[+] PROCESSING BUCK FILES...0.1s
BUILD FAILED: Couldn't get dependency '//.okbuck/keystore:app_keystore' of target '//app:bin':
The build file that should contain //.okbuck/keystore:app_keystore has already been parsed (/Users/hahack/application.okbuck/keystore/BUCK), but //.okbuck/keystore:app_keystore was not found. Please make sure that //.okbuck/keystore:app_keystore is defined in /Users/hahack/application.okbuck/keystore/BUCK.
```

Here's my root build.gradle:

```
apply plugin: 'com.github.piasy.okbuck-gradle-plugin' // facebook buck
okbuck {
    target "android-20"
    overwrite true
    resPackages = [
        'app': 'com.hahack.app'
    ]
}
```

And the content of .okbuck/app/BUCK looks like this:

``` py
import re

jar_deps = []
for jarfile in glob(['*.jar']):
	name = 'jars__' + re.sub(r'^.*/([^/]+)\.jar$', r'\1', jarfile)
	jar_deps.append(':' + name)
	prebuilt_jar(
		name = name,
		binary_jar = jarfile,
	)

android_library(
	name = 'all-jars',
	exported_deps = jar_deps,
	visibility = [
		'PUBLIC',
	],
)

aar_deps = []
for aarfile in glob(['*.aar']):
	name = 'aar__' + re.sub(r'^.*/([^/]+)\.aar$', r'\1', aarfile)
	aar_deps.append(':' + name)
	android_prebuilt_aar(
		name = name,
		aar = aarfile,
	)

android_library(
	name = 'all-aars',
	exported_deps = aar_deps,
	visibility = [
		'PUBLIC',
	],
)
```

**Comment by Piasy** [comment]
That looks strange, could you please save `./gradlew okbuck --stacktrace`'s output to a file and attach it in a comment? The file structure of your project, and the content of `app/BUCK` (**not `.okbuck/app/BUCK`**) could also helpful to figure out the cause of this problem. Thanks!

**Comment by wzpan** [comment]
Hi @Piasy ,

Thanks for the rapid response.

> could you please save ./gradlew okbuck --stacktrace's output to a file and attach it in a comment?

Something went wrong that I can't attach the log file to the comment. Thus I will send you a mail with the attachment. Please check it out. Thanks again.

**Comment by Piasy** [comment]
@wzpan Hi, I didn't receive your email after half hour... is there something wrong? Or maybe you can contact me with QQ: 200842165. Thanks!

**Comment by wzpan** [comment]
Hi @Piasy ,

Sorry I went to have dinner with my friends. Now you should have got the email! Thanks.

**Comment by Piasy** [comment]
@wzpan Hi, I can reproduce your problem with version 0.2.5, which didn't support multiple application module, a newer version 0.2.6 fix it, so could you please give version 0.2.6 a trial? Thanks!

**Comment by wzpan** [comment]
Ok. I'll try it this afternoon.

**Comment by wzpan** [comment]
Hi @Piasy ,

I switch to use 0.2.6 and now I come across another problem.

For application module, I got the following error:

```
➜  appframework git:(master) ✗ buck run hfcache_example
Using watchman.
Using buckd.
[+] PROCESSING BUCK FILES...0.6s [100%]
BUILD FAILED: Couldn't get dependency '//hfasynchttp:res' of target '//hfcache_example:src':
//hfasynchttp:res: parameter 'res': no such file or directory 'hfasynchttp/src/main/res'
```

And for library module, I got the following error:

```
➜  appframework git:(master) ✗ buck run hfcache        
Using buckd.
[+] PROCESSING BUCK FILES...0.2s [100%]
BUILD FAILED: No rule found when resolving target //hfcache:hfcache in build file //hfcache/BUCK
Defined in file: /Users/hahack/hyperion/hyperion-sdk/afengine/android/framework/appframework/hfcache/BUCK
```

**Comment by Piasy** [comment]
@wzpan Well, I must admit that your project is a really big one, with many modules, which I didn't take into consideration when develop OkBuck. So I can imagine that there must be more problems waiting for you, and me too. 

To address your problems, and improve OkBuck, more efficiently, would you mind to make contact with me through some IM tools? e.g. QQ: `200842165`, WeChat: `piasy_umumu`.

**Comment by Piasy** [comment]
In summary, there are three problems:
1. If an android library module doesn't have a res folder, OkBuck will generate the wrong BUCK file, and this will be fixed soon;
2. If your project is under other version control system than git, e.g. git-svn, BUCK will try to recognize files under these folders, which should be ignored, and I'll add `.svn` ignore support in OkBuck soon, but I  still suggest you use git :) ;
3. BuildConfig problems, custom build config should be defined inside `defaultConfig` block of build.gradle file, otherwise, OkBuck and BUCK will ignore these custom fields.

**Comment by Piasy** [comment]
Above problems is fixed by 09e9f6e2ba27c1ab2fe7c85eae13370890378ff0 and c30fd6802db2d7fb7f60a806a5fe9c2dfc5c00ce.
